mungegithub/mungers/submit-queue.go

```diff
@@ -159,8 +160,14 @@ type submitQueueInterruptedObject struct {
 type submitQueueMetadata struct {
 	ProjectName string
 
-	ChartUrl    string
-	HistoryUrl  string
+	ChartUrl   string
+	HistoryUrl string
+	// chartUrl and historyUrl are option storage locations. They are distinct from ChartUrl and
+	// HistoryUrl since the the public variables are used asynchronously by a fileserver and updates
+	// to the options values should not cause a race condition.
```

Comment:
It's *probably* in a single cacheline, but unless the program specifically reads the 16B value in one instruction, you're right, it could race.

Your caution is good. I was hoping we could avoid some of the ugliness, but that doesn't look like it's possible. :-)

More details here: https://github.com/golang/go/issues/9816#issuecomment-73562437